crates/sui/tests/ptb_files_tests.rs

```diff
@@ -9,84 +9,93 @@ use sui::ptb::{
 };
 use test_cluster::TestClusterBuilder;
 
-// check if all good ptb files can be parsed OK
-fn test_parse_all_ptb_files(path: &Path) -> datatest_stable::Result<()> {
+const TEST_DIR: &str = "tests";
+
+#[cfg_attr(not(msim), tokio::main)]
+async fn test_ptb_files(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+    std::env::set_var("NO_COLOR", "true"); // we need this for the miette errors
     let ptb = PTB::default();
     let cmd = PTB::command();
     let file = path.to_str().unwrap();
-    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
+    let args = cmd.get_matches_from(vec!["ptb", "--file", file, "--preview"]);
     let cwd = std::env::current_dir().unwrap();
     let commands = ptb.from_matches(cwd, &args, &mut BTreeMap::new()).unwrap();
 
+    // === PREVIEW ===
+    let ptb_preview = ptb.preview(&commands);
+    let mut results = vec![];
+    let preview_string = if let Some(ptb_preview) = ptb_preview {
+        ptb_preview.to_string()
+    } else {
+        "".to_string()
+    };
+    results.push(" === PREVIEW === ".to_string());
+    results.push(preview_string);
+
+    // === PARSE COMMANDS ===
     let mut parser = PTBParser::new();
-    for (_, cmd) in commands {
+    for (_, cmd) in &commands {
         parser.parse(cmd.clone());
     }
-    let (_parsed_commands, errors) = parser.finish();
-    assert!(errors.is_empty());
-
-    Ok(())
-}
+    let (parsed, errors) = parser.finish();
 
-// check if the bad ptb files return errors when parsed
-fn test_parse_bad_ptb_files(path: &Path) -> datatest_stable::Result<()> {
-    let ptb = PTB::default();
-    let cmd = PTB::command();
-    let file = path.to_str().unwrap();
-    let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
-    let cwd = std::env::current_dir().unwrap();
-    let commands = ptb.from_matches(cwd, &args, &mut BTreeMap::new()).unwrap();
-    let mut parser = PTBParser::new();
-    for (_, cmd) in commands.clone() {
-        parser.parse(cmd.clone());
+    results.push(" === Parsed PTB === ".to_string());
+    for c in &parsed {
+        let values = c
+            .args
+            .iter()
+            .map(|x| x.value.to_string())
+            .collect::<Vec<_>>();
+        results.push(format!(
+            "cmd: {}, value: {:?}",
+            c.name.value.to_string(),
+            values
+        ));
     }
-    let (_parsed_commands, errors) = parser.finish();
-    let rendered = render_errors(commands, errors.clone());
-    for e in rendered.iter() {
-        println!("{:?}", e);
+
+    if !errors.is_empty() {
+        let rendered = render_errors(commands.clone(), errors);
+        results.push(" === PARSING PTB ERRORS === ".to_string());
+        for e in rendered.iter() {
+            results.push(format!("{:?}", e));
+        }
+        return Ok(());
```

Comment:
This doesn't seem right -- if the parser has errors we won't save out the errors to the insta snapshot it seems? 

I think we want to do the `parse_and_build_ptb` step (and the results of the build/errors can be added to the string inside of the `if`) in this case, but if the parsing errors we still do want to save those out to the insta snapshots.

Reply:
Yes, indeed, sorry about that. My mind is elsewhere.

Reply:
I think I fixed it now.